server/src/main/java/org/apache/druid/server/security/ForbiddenException.java

```diff
@@ -44,4 +48,14 @@ public String getErrorMessage()
   {
     return super.getMessage();
   }
+
+  @Override
+  public ForbiddenException transform(Function<String, String> errorMessageTransformFunction)
+  {
+    if (Strings.isNullOrEmpty(errorMessageTransformFunction.apply(getMessage()))) {
+      return new ForbiddenException();
+    } else {
+      return this;
```

Comment:
appears to be a logic bug here

unit tests please

Reply:
Done